adminSiteClient/GdocsPreviewPage.tsx

```diff
@@ -31,61 +32,114 @@ import { GdocsEditLink } from "./GdocsEditLink.js"
 import { openSuccessNotification } from "./gdocsNotifications.js"
 import { GdocsDiffButton } from "./GdocsDiffButton.js"
 import { GdocsDiff } from "./GdocsDiff.js"
+import { useInterval } from "../site/hooks.js"
 
 export const GdocsPreviewPage = ({ match, history }: GdocsMatchProps) => {
     const { id } = match.params
-    const [gdoc, setGdoc] = useState<OwidArticleType>()
-    const [originalGdoc, setOriginalGdoc] = useState<OwidArticleType>()
+    const [gdoc, setGdoc] = useState<{
+        original?: OwidArticleType
+        current?: OwidArticleType
+    }>({ original: undefined, current: undefined })
+    const originalGdoc = gdoc.original
+    const currentGdoc = gdoc.current
+    const setOriginalGdoc = (original?: OwidArticleType) =>
+        setGdoc({ original, current: gdoc.current })
+    const setCurrentGdoc = (current: OwidArticleType | undefined) =>
+        setGdoc({ original: gdoc.original, current })
+    const hasChanges = useGdocsChanged(originalGdoc, currentGdoc)
     const [isSettingsOpen, setSettingsOpen] = useState(false)
+    const [hasSyncingError, setHasSyncingError] = useState(false)
+    const [criticalErrorMessage, setCriticalErrorMessage] = useState<
+        undefined | string
+    >()
     const [isDiffOpen, setDiffOpen] = useState(false)
     const [errors, setErrors] = React.useState<ErrorMessage[]>()
-
     const { admin } = useContext(AdminAppContext)
     const store = useGdocsStore()
 
-    const hasChanges = useGdocsChanged(originalGdoc, gdoc)
-    const syncingError = useUpdatePreviewContent(id, gdoc, setGdoc, admin)
-    const isLightningUpdate = useLightningUpdate(originalGdoc, gdoc, hasChanges)
-    useAutoSaveDraft(gdoc, setOriginalGdoc, hasChanges)
+    const fetchGdoc = useCallback(
```

Comment:
@mlbrgl I made these changes because race conditions were emerging on the intial load. Tracking state and responsibilities between this component and the `useUpdatePreviewContent` hook was getting tricky to debug so I lifted everything back into this component. Hope it makes sense to you!